static/js/composebox_typeahead.js

```diff
@@ -369,11 +368,11 @@ function get_mention_candidates_data(is_silent) {
                 full_name: mention,
             };
         });
-        groups = user_groups.get_realm_user_groups();
     }
 
+    var groups = user_groups.get_realm_user_groups();
     var persons = people.get_realm_persons();
-    return [].concat(persons, all_items, groups);
+    return [].concat(persons, groups, all_items);
```

Comment:
Why did you change this line?  It seems like this might change the ordering of results, and seeing unrelated to the changes you were making...

Reply:
Since I've changed the order in which the concatenation of mention items (people, group, etc.) is done in node test (https://github.com/zulip/zulip/pull/11862/files#diff-f183345766f158f180b91d64e33b91e9R1183), I have to change the order here as well otherwise tests were failing.

Reply:
We shouldn't alter the actual code for the tests in this case. I do agree with the order your changes bring (keeping all items at the end) but that should be a separate change, if we do decide to have it.

Reply:
@timabbott Should we keep the current order (all item at the end)?

Reply:
@overide I think it's fine to change the order to move the all items to the end, but that should be its own commit (you can insert it before this one).  

Can you please split that out?